File: src/statesman/commands/InventoryConditionalCommand.java
package statesman.commands;

import statesman.Interpreter;

public class InventoryConditionalCommand extends ConditionalCommand {

    public static final String ID = "invcond";

    private String[] _itemNames;

    public InventoryConditionalCommand() {
        super();
    }

    public InventoryConditionalCommand(
            CommandGroup group,
            CommandGroup elseGroup,
            String[] itemNames,
            boolean[] targetValues,
            boolean orMode) {
        super();
        _group = group;
        _elseGroup = elseGroup;
        _itemNames = itemNames;
        _targetValues = targetValues;
        _orMode = orMode;
    }

    @Override
    public Command createInstance(String[] arguments) {
        if (arguments.length == 2) {
            String condition = arguments[1];
            boolean orMode = useOrOperator(condition);
            String[] parts = getConditionParts(condition, orMode);
            boolean[] targetValues = new boolean[parts.length];
            String[] itemNames = new String[parts.length];
            
            for (int i = 0; i < parts.length; i++) {
                targetValues[i] = !parts[i].startsWith("!");
                if (targetValues[i]) {
                    itemNames[i] = parts[i];
                } else {
                    itemNames[i] = parts[i].substring(1);
                }
            }
            
            return new InventoryConditionalCommand(
                    new CommandGroup(""),
                    new CommandGroup(""),
                    itemNames,
                    targetValues,
                    orMode);
        }
        
        return null;
    }

    @Override
    public void execute() {
        for (int i = 0; i < _itemNames.length; i++) {
            boolean currentState = false;
            currentState =
                    (Interpreter.getInventory().containsKey(_itemNames[i]) == _targetValues[i]);
            boolean stopLooping = updateState(currentState); 
            if (stopLooping) {
                break;
            }
        }
        super.execute();
    }

    public String[] getItemNames() {
        return _itemNames;
    }

}

File: src/Commands/InventoryConditionalCommand.cs
namespace Statesman.Commands
{
    public class InventoryConditionalCommand : ConditionalCommand
    {
        public static readonly string ID = "invcond";

        public string[] ItemNames { get; }

        public InventoryConditionalCommand()
            : base()
        {
        }

        public InventoryConditionalCommand(
                CommandGroup group,
                CommandGroup elseGroup,
                string[] itemNames,
                bool[] targetValues,
                bool orMode)
            : base()
        {
            _group = group;
            _elseGroup = elseGroup;
            ItemNames = itemNames;
            _targetValues = targetValues;
            _orMode = orMode;
        }

        public override Command CreateInstance(string[] arguments)
        {
            if (arguments.Length == 2)
            {
                string condition = arguments[1];
                bool orMode = UseOrOperator(condition);
                string[] parts = GetConditionParts(condition, orMode);
                bool[] targetValues = new bool[parts.Length];
                string[] itemNames = new string[parts.Length];

                for (int i = 0; i < parts.Length; i++)
                {
                    targetValues[i] = !parts[i].StartsWith("!");
                    if (targetValues[i])
                    {
                        itemNames[i] = parts[i];
                    }
                    else
                    {
                        itemNames[i] = parts[i].Substring(1);
                    }
                }

                return new InventoryConditionalCommand(
                        new CommandGroup(""),
                        new CommandGroup(""),
                        itemNames,
                        targetValues,
                        orMode);
            }

            return null;
        }

        public override void Execute()
        {
            for (int i = 0; i < ItemNames.Length; i++)
            {
                bool currentState =
                    Interpreter.Inventory.ContainsKey(ItemNames[i]) == _targetValues[i];
                bool stopLooping = UpdateState(currentState);
                if (stopLooping)
                {
                    break;
                }
            }
            base.Execute();
        }
    }
}
